scheduler/src/main/java/com/flipkart/flux/clientelb/dao/ClientElbDAOImpl.java

```diff
@@ -0,0 +1,126 @@
+import com.flipkart.flux.persistence.Storage;
+import com.google.inject.name.Named;
+import org.hibernate.Criteria;
+import org.hibernate.Query;
+import org.hibernate.Session;
+import org.hibernate.criterion.Order;
+import org.hibernate.criterion.Restrictions;
+
+import javax.inject.Inject;
+import javax.transaction.Transactional;
+import java.util.List;
+
+/**
+ * <code>ClientElbDAOImpl</code> is an implementation of {@link ClientElbDAO} which uses Hibernate to perform
+ *  CRUD operations.
+ *
+ * @author akif.khan
+ */
+public class ClientElbDAOImpl implements ClientElbDAO {
+
+    private SessionFactoryContext sessionFactoryContext;
+
+    @Inject
+    public ClientElbDAOImpl(@Named("schedulerSessionFactoriesContext") SessionFactoryContext sessionFactoryContext) {
+        this.sessionFactoryContext = sessionFactoryContext;
+    }
+
+    /**
+     * Creates client elb entry in DB and returns saved object
+     */
+    @Override
+    @Transactional
+    @SelectDataSource(storage = Storage.SCHEDULER)
+    public ClientElb create(String clientElbId, ClientElb clientElb) {
+        currentSession().save(clientElb);
+        return clientElb;
+    }
+
+    /**
+     * Retrieves Client Elb from DB by it's unique identifier
+     */
+    @Override
+    @Transactional
+    @SelectDataSource(storage = Storage.SCHEDULER)
+   public  ClientElb findById(String id) {
+        Criteria criteria = currentSession().createCriteria(ClientElb.class)
+                .add(Restrictions.eq("id", id));
+        Object object = criteria.uniqueResult();
+        ClientElb castedObject = null;
+        if(object != null)
+            castedObject = (ClientElb) object;
+        else
+            castedObject = null;
+        return castedObject;
+    }
+
+    /**
+     * Updates Elb Url of a client in DB
+     */
+    @Override
+    @Transactional
+    @SelectDataSource(storage = Storage.SCHEDULER)
+    public void updateElbUrl(String clientElbId, String clientElbUrl) {
+        Query updateQuery = currentSession().createQuery(
+                "update ClientElb set elbUrl = :clientElbUrl where id = :clientElbId");
+        updateQuery.setString("clientElbUrl", clientElbUrl != null ? clientElbUrl.toString() : null);
+        updateQuery.setString("clientElbId", clientElbId);
+        updateQuery.executeUpdate();
+    }
+
+    /**
+     * Delete Client Elb entry from DB
+     */
+    @Override
+    @Transactional
+    @SelectDataSource(storage = Storage.SCHEDULER)
+    public void delete(String id) {
+        Query deleteQuery = currentSession().createQuery("delete ClientElb c where c.id=:id");
+        deleteQuery.setString("id", id);
+        deleteQuery.executeUpdate();
+    }
+
+    /**
+     * Retrieve Older Client Elb entry limit by parameter rowCount
+     * @param rowCount
+     * @return List of oldest ClientElbs'
+     */
+    @Override
+    @Transactional
+    @SelectDataSource(storage = Storage.SCHEDULER)
```

Comment:
We don't need refreshing policy.

Reply:
Added this retrieveOldest to retrieve oldest entry sorted by creation time upto a specified count to support LRU policy in customized cache.
Since, we are using now Guava Cache, don't need this function anymore, removing it.
Also, as per our use case, cache won't need rotation policy.